mitiq/factories.py

```diff
@@ -0,0 +1,129 @@
+from pyquil import Program
+from typing import List, Tuple, Callable
+
+import mitiq.qiskit.qiskit_utils as qs_utils
+import numpy as np
+
+class Factory(object):
```

Comment:
Maybe `BaseFactor` or `AbstractFactor`? I think either or just `Factory` is fine, just a suggestion.

Reply:
I have also many comments-ideas-suggestions about Factories. So I think it is better if we discuss all these open questions to a new issue that I will create soon.